### PR TITLE
Added serde derived Serializer behind a feature flag

### PR DESCRIPTION
**Pull request by josephhajduk**
Did not add it to builders or error types.

This was useful for me, thanks for the very useful library.  I made a similar PR to sdp-types

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -16,3 +16,7 @@ nom = "7.0"
 cookie-factory = "0.3"
 tinyvec = {version = "1.0", features = ["std", "rustc_1_57"]}
 url = "2.0"
+
+serde = { version = "1.0", optional = true, features = ["derive"] }
+[features]
+serde_serialize = ["serde"]
```
Comment (sdroege) [inline Cargo.toml:22]: I think this feature is usually called `serde`

File: src/message_ref.rs [hunk 2]
```diff
@@ -210,21 +214,38 @@ impl<'a> RequestRef<'a> {
...
     pub(crate) version: Version,
     pub(crate) status: StatusCode,
     pub(crate) reason_phrase: &'a str,
+    #[cfg_attr(feature = "serde_serialize", serde(serialize_with = "serialize_tinyvec"))]
     pub(crate) headers: TinyVec<[HeaderRef<'a>; 16]>,
     pub(crate) body: &'a [u8],
 }
 
+#[cfg(feature = "serde_serialize")]
+fn serialize_tinyvec<S, T>(tinyvec: &TinyVec<[T; 16]>, serializer: S) -> Result<S::Ok, S::Error>
```
Comment (sdroege) [inline src/message_ref.rs:236]: If you enable the `serde` feature on `tinyvec` then this shouldn't be necessary

**Review by sdroege — COMMENTED** [review]
Why only `Serialize` and not also `Deserialize` while we're at it? How are you using this? :)

File: Cargo.toml (hunk 1 above)
Comment (josephhajduk) [reply Cargo.toml:22]: I can rename and add both serialize and deserialize

File: src/message_ref.rs (hunk 2 above)
Comment (josephhajduk) [reply src/message_ref.rs:236]: I will make this change, I didn't bother checking.  I will make the changes.

**Comment by josephhajduk** [comment]
I didn't need Deserialize but it is easy enough to add it,  I am using it for tracing/logging.

**Comment by sdroege** [comment]
I see, that makes sense

**Comment by sdroege** [comment]
```
  namespaced features with the `dep:` prefix are only allowed on the nightly channel and requires the `-Z namespaced-features` flag on the command-line
```

That's OK, let's update the MSRV to something more reasonable that supports this :) I'll do that, don't worry about it.

**Comment by josephhajduk** [comment]
Great thanks!

**Review by sdroege — APPROVED** [review]
Thanks!

**Comment by sdroege** [comment]
> Did not add it to builders or error types.

OOC, why did you add it to error types in `sdp-types` but not here?

**Comment by josephhajduk** [comment]
I was wildly pasting in sdp-types and didn't think it through, I don't think it makes sense either I will remove it

**Pull request merged**